src/CodeGen_Vulkan_Dev.cpp

```diff
@@ -0,0 +1,2928 @@
+        internal_assert(op->args.size() == 2);
+        if (op->type.is_uint() || (op->args[1].type().is_uint())) {
+            visit_binary_op(SpvOpShiftRightLogical, op->type, op->args[0], op->args[1]);
+        } else {
+            Expr e = lower_signed_shift_right(op->args[0], op->args[1]);
+            e.accept(this);
+        }
+    } else if (op->is_intrinsic(Call::shift_left)) {
+        internal_assert(op->args.size() == 2);
+        if (op->type.is_uint() || (op->args[1].type().is_uint())) {
+            visit_binary_op(SpvOpShiftLeftLogical, op->type, op->args[0], op->args[1]);
+        } else {
+            Expr e = lower_signed_shift_left(op->args[0], op->args[1]);
+            e.accept(this);
+        }
+    } else if (op->is_intrinsic(Call::strict_float)) {
+        // TODO: Enable/Disable RelaxedPrecision flags?
+        internal_assert(op->args.size() == 1);
+        op->args[0].accept(this);
+    } else if (op->is_intrinsic(Call::IntrinsicOp::sorted_avg)) {
+        internal_assert(op->args.size() == 2);
+        // b > a, so the following works without widening:
+        // a + (b - a)/2
+        Expr e = op->args[0] + (op->args[1] - op->args[0]) / 2;
+        e.accept(this);
+    } else if (op->is_intrinsic(Call::lerp)) {
+
+        // Implement lerp using GLSL's mix() function, which always uses
+        // floating point arithmetic.
+        Expr zero_val = op->args[0];
+        Expr one_val = op->args[1];
+        Expr weight = op->args[2];
+
+        internal_assert(weight.type().is_uint() || weight.type().is_float());
+        if (weight.type().is_uint()) {
+            // Normalize integer weights to [0.0f, 1.0f] range.
+            internal_assert(weight.type().bits() < 32);
+            weight = Div::make(Cast::make(Float(32), weight),
+                               Cast::make(Float(32), weight.type().max()));
+        } else if (op->type.is_uint()) {
+            // Round float weights down to next multiple of (1/op->type.imax())
+            // to give same results as lerp based on integer arithmetic.
+            internal_assert(op->type.bits() < 32);
+            weight = floor(weight * op->type.max()) / op->type.max();
+        }
+
+        Type result_type = Float(32, op->type.lanes());
+        Expr e = Call::make(result_type, "mix", {zero_val, one_val, weight}, Call::Extern);
+
+        if (!op->type.is_float()) {
+            // Mirror rounding implementation of Halide's integer lerp.
+            e = Cast::make(op->type, floor(e + 0.5f));
+        }
+        e.accept(this);
+
+    } else if (op->is_intrinsic(Call::mux)) {
+        Expr e = lower_mux(op);
+        e.accept(this);
+    } else if (op->is_intrinsic(Call::saturating_cast)) {
+        Expr e = lower_intrinsic(op);
+        e.accept(this);
+
+    } else if (op->is_intrinsic()) {
+        Expr lowered = lower_intrinsic(op);
+        if (lowered.defined()) {
+            Expr e = lower_intrinsic(op);
+            e.accept(this);
```

Comment:
I think this should be:
```
lowered.accept(this);
```

Reply:
Oh geez ... yeah, definitely not what I intended! Thanks for catching this.  I'll fix this!